### PR TITLE
Update keyvault patch release september 2020

### PR DESCRIPTION
**Pull request by vcolin7**
Bumped versions and updated CHANGELOG and README files.

### CODE REVIEW

**Review by alzimmermsft — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-encryption/pom.xml [hunk 1]
```diff
@@ -49,7 +49,7 @@ Licensed under the MIT License.
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-security-keyvault-keys</artifactId>
-      <version>4.3.0-beta.1</version> <!-- {x-version-update;com.azure:azure-security-keyvault-keys;current} -->
+      <version>4.2.1</version> <!-- {x-version-update;com.azure:azure-security-keyvault-keys;current} -->
```
Comment (samvaity) [inline sdk/cosmos/azure-cosmos-encryption/pom.xml:52]: @kushagraThapar could you confirm if this needs to be a current dependency or dependency version should work?

File: sdk/storage/azure-storage-blob-cryptography/pom.xml [hunk 2]
```diff
@@ -109,7 +109,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-security-keyvault-keys</artifactId>
-      <version>4.3.0-beta.1</version> <!-- {x-version-update;unreleased_com.azure:azure-security-keyvault-keys;dependency} -->
+      <version>4.2.1</version> <!-- {x-version-update;unreleased_com.azure:azure-security-keyvault-keys;dependency} -->
```
Comment (samvaity) [inline sdk/storage/azure-storage-blob-cryptography/pom.xml:112]: @vcolin7 Could you confirm if we are releasing key beta with `4.3.0-beta.1` if yes, then we should not change this.

File: sdk/cosmos/azure-cosmos-encryption/pom.xml (hunk 1 above)
Comment (vcolin7) [reply sdk/cosmos/azure-cosmos-encryption/pom.xml:52]: This is just for the from source pipeline to work. This change won't go into master.

File: sdk/storage/azure-storage-blob-cryptography/pom.xml (hunk 2 above)
Comment (vcolin7) [reply sdk/storage/azure-storage-blob-cryptography/pom.xml:112]: This is just for the from source pipeline to work. This change won't go into master.

File: sdk/cosmos/azure-cosmos-encryption/pom.xml (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-encryption/pom.xml:52]: as long as cosmos `Encryption_Integration_Tests_Java8` test pass we are fine

**Pull request merged**